### PR TITLE
geometry 2.0.0 (new formula)

### PR DESCRIPTION
**Pull request by danielbayley**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/geometry.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  sha256 "47c9fd345e5e8f093ae735a91fe5a9a9a6e3c1873eab220a7846a17b8405de02"
+  license "ISC"
+  head "https://github.com/geometry-zsh/geometry.git", branch: "main"
+
+  depends_on "zsh-async"
+  uses_from_macos "expect" => :test
+  uses_from_macos "zsh" => :test
+
+  def install
+    libexec.install ["functions", "geometry.zsh"]
```
Comment (carlocab) [inline Formula/geometry.rb:14]:
```suggestion
    pkgshare.install ["functions", "geometry.zsh"]
```
This seems more typical of other Zsh plugins.

File: Formula/geometry.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def caveats
+    <<~EOS
+      To activate Geometry, add the following to your .zshrc:
+        source #{libexec}/geometry.zsh
```
Comment (carlocab) [inline Formula/geometry.rb:20]:
```suggestion
        source #{opt_pkgshare}/geometry.zsh
```

File: Formula/geometry.rb (hunk 1 above)
Comment (danielbayley) [reply Formula/geometry.rb:14]: Really, all `prompt` themes should be `zsh_function.install`ed so they can be `autoload`ed for `promptinit`, but this one doesn't work…

**Comment by chenrui333** [comment]
looks like the test is still failing for 12

**Comment by danielbayley** [comment]
> looks like `test` is still failing for `12`

@chenrui333 So strange, since that’s the version I am actually still using on my local machine, where the `test` passes…

The same test was passing before changing `libexec` to [`opt_`]`pkgshare`, as suggested by @carlocab. How could this affect `expect`?

**Comment by danielbayley** [comment]
@chenrui333 Passing now…?! 🤷🏻‍♂️

**Review by chenrui333 — DISMISSED** [review]

File: Formula/geometry.rb (hunk 1 above)
Comment (carlocab) [reply Formula/geometry.rb]: Yes, I don't find `pkgshare` a great place either (some `zsh_*` location is probably better), but that's where a bunch of other formula install `*.zsh` plugins, so we should at least try to be consistent.

**Review by carlocab — DISMISSED** [review]

**Comment by carlocab** [comment]
Odd... We just have a bunch of text files. I wonder if we've broken bottle reproducibility somewhere.
```
❯ sha256sum *.tar.gz
348594557bfc3c624b36587813a25116a7c8c33105f31c8575dd4ef929313c1a  geometry--2.0.0.arm64_big_sur.bottle.tar.gz
480610e8e542d736504fa4d739d2eb46cb14e398824968422dcbb878721707a9  geometry--2.0.0.arm64_monterey.bottle.tar.gz
3cb5530758212e247046b03b8b0320ffcfb24b7afcca9656e792e490335ec54f  geometry--2.0.0.arm64_ventura.bottle.tar.gz
524875fa324bbd0e9046def6c3348beaf35eb2c5f3e3ed46152f04660065cd7f  geometry--2.0.0.big_sur.bottle.tar.gz
47c45998f48eb2485993834cf79861c6f7b02fb380765351b7d374a3568d5c8a  geometry--2.0.0.monterey.bottle.tar.gz
788cbccf9fc8997ef005203f9df33567e8880618df892fae6209cf7272cc3b3c  geometry--2.0.0.ventura.bottle.tar.gz
77d688cd33571e3f3fcdd38e1b49fb2ed6188445a2ac9db527d6093b694528d3  geometry--2.0.0.x86_64_linux.bottle.tar.gz
```

CC @Homebrew/brew

**Comment by Bo98** [comment]
Yep, looks broken by https://github.com/Homebrew/brew/pull/14479. `require` should be at the end I think.

File: Formula/geometry.rb (hunk 1 above)
Comment (danielbayley) [reply Formula/geometry.rb]: > we should at least try to be consistent

Agreed, but looking at other `prompt` formulae, it is currently very inconsistent… Some of them use `#{HOMEBREW_PREFIX}/share/` (would `opt` not be more appropriate here, anyway?), others `#{opt_prefix}/`, others [`opt_`]`pkgshare/`, and others still `zsh_function`.

Some have `caveats`, but others do not… It all seems a bit arbitrary.

For `zsh` `prompt`s specifically, `autoload`ing them from `#{prefix}/share/zsh/site-functions` (the `zsh_function.install` location, already found on`FPATH`) is the _proper_ way to install a `prompt` theme, and should speed up `zsh`ell initial load time, as opposed to just `source`ing directly in `.zshrc`…

**Review by carlocab — APPROVED** [review]
Rebased to pull in the bottle fix. Can merge once CI is complete.

**Comment by p-linnane** [comment]
I'm not sure this should have been added to `homebrew-core`, as it appears to be unmaintained. Livecheck is returning 2.2.0, [so I opened a PR](https://github.com/Homebrew/homebrew-core/pull/123229) to bump this. Turns out the last commit was July 2021, and the latest release (2.0.0) was August 2019.

**Comment by carlocab** [comment]
This is just a shell plugin, so the need for maintenance is quite low. I've seen widely used completion scripts that haven't changed for much, much longer than that.

**Comment by p-linnane** [comment]
Are we ok with changing to `strategy :github_latest` for livecheck and calling it a day then?

**Comment by carlocab** [comment]
We can probably bump this to 2.2.0 actually. It looks like only major releases get registered as GitHub releases, but later tags seem fine for general use.

**Comment by p-linnane** [comment]
@carlocab Would you mind approving my linked PR so we can get it merged?

**Pull request closed without merging**